### PR TITLE
APM > .NET > Logs injection

### PR DESCRIPTION
**Pull request by lucaspimentel**
### What does this PR do?
- "Connect Logs and Traces" page
  - add .NET tabs
- "Tracing .NET Applications" page
  - add new settings for log injection to settings table
  - add text to all settings tables to clarify what the columns mean (settings vs property)

### Motivation
We released [.NET Tracer 1.4](https://github.com/DataDog/dd-trace-dotnet/releases/tag/v1.4.0) with support for injecting trace ids into logs.

### Preview links
- [Connect Logs and Traces](https://docs-staging.datadoghq.com/lucas.p/apm-dotnet-logs/tracing/advanced/connect_logs_and_traces/?tab=net) (new .NET tabs)
- [Tracing .NET Applications](https://docs-staging.datadoghq.com/lucas.p/apm-dotnet-logs/tracing/languages/dotnet/)

### CODE REVIEW

File: content/en/tracing/advanced/connect_logs_and_traces.md [hunk 1]
```diff
@@ -119,6 +119,26 @@ This enables automatic trace ID injection for `winston`, `bunyan`, and `pino`.
 
 **Note**: Automatic injection only works for logs formatted as JSON.
 
+{{% /tab %}}
+{{% tab ".NET" %}}
+
+Enable injection in the .NET Tracer’s [configuration][1] by setting `DD_LOGS_INJECTION=true` through environment variables or configuration files, or in code:
```
Comment (zacharycmontoya) [inline content/en/tracing/advanced/connect_logs_and_traces.md]: Currently, it is only turned on at startup of the tracer. So for now, this only works by setting `DD_LOGS_INJECTION=true.` We can create a backlog item to dynamically add it later in execution by setting the `Tracer.Instance.Settings.LogsInjectionEnabled = true;` but that does not work right now.

Comment (lucaspimentel) [reply content/en/tracing/advanced/connect_logs_and_traces.md]: Oh, right. We talked about this. 🤦‍♂

Comment (lucaspimentel) [reply content/en/tracing/advanced/connect_logs_and_traces.md]: I changed the code sample and added text to clarify that later changes are ignored.

File: content/en/tracing/advanced/connect_logs_and_traces.md [hunk 2]
```diff
@@ -119,6 +119,26 @@ This enables automatic trace ID injection for `winston`, `bunyan`, and `pino`.
...
+
+```csharp
+using Datadog.Trace;
+
+Tracer.Instance.Settings.LogsInjectionEnabled = true;
+```
+
+The .NET Tracer uses the [LibLog][2] library to integration into your application logs. It contains transparent built-in support for [NLog][3], [Log4Net][4], [Serilog][5] and [Loupe][6].
```
Comment (zacharycmontoya) [inline content/en/tracing/advanced/connect_logs_and_traces.md]: While LibLog supports Loupe, Loupe does not support the Mdc feature (Mapped Diagnostics Context) feature, so we can't instrument that logging framework. We should somehow note that Loupe will not be automatically instrumented.

File: content/en/tracing/advanced/connect_logs_and_traces.md [hunk 3]
```diff
@@ -119,6 +119,30 @@ This enables automatic trace ID injection for `winston`, `bunyan`, and `pino`.
...
+
+```csharp
+using Datadog.Trace;
+using Datadog.Trace.Configuration;
+
+var settings = new TracerSettings { LogsInjectionEnabled = true };
+var tracer = new Tracer(settings);
+```
+
+This settings is only read during `Tracer` initialization. Changes to this setting after the `Tracer` instance is created are ignored.
```
Comment (zacharycmontoya) [inline content/en/tracing/advanced/connect_logs_and_traces.md]: typo: "setting"

File: content/en/tracing/advanced/connect_logs_and_traces.md (hunk 2 above)
Comment (zacharycmontoya) [reply content/en/tracing/advanced/connect_logs_and_traces.md]: The Java page currently has this disclaimer. Perhaps we could do something similar:

**Note**: Currently only **slf4j** is supported for MDC autoinjection.

File: content/en/tracing/setup/dotnet.md [hunk 4]
```diff
@@ -286,21 +287,20 @@ ENV DD_INTEGRATIONS=/opt/datadog/integrations.json
...
-`DD_DISABLED_INTEGRATIONS`               | `DisabledIntegrationNames` | Sets a list of integrations to disable. All other integrations remain enabled. If not set, all integrations are enabled. Supports multiple values separated with semicolons. Valid values are the integration names listed in the [Integrations][3] section below. |
-`DD_TRACE_ANALYTICS_ENABLED`             | `AnalyticsEnabled`         | Shorthand that enables default Trace Search and Analytics settings for web framework integrations. Valid values are: `true` or `false` (default).                                                                                                                  |
+The following column lists integration-specific settings. The first column indicates the name used in environment variables or configuration files. The second column indicates the name of the propery on the `IntegrationSettings` class. You can access these properties in code through `Tracer.Instance.Settings.Integrations["<INTEGRATION>"]`. Integration names are listed in the [Integrations][3] section below.
```
Comment (zacharycmontoya) [inline content/en/tracing/setup/dotnet.md]: "The following table"

**Comment by zacharycmontoya** [comment]
Those are all the comments I have right now. Let's see what Andrew says

File: content/en/tracing/advanced/connect_logs_and_traces.md (hunk 2 above)
Comment (lucaspimentel) [reply content/en/tracing/advanced/connect_logs_and_traces.md]: I removed the reference to Loupe in 5e09060cda4945368f8bf6d890f5f1f04740c92e.

File: content/en/tracing/advanced/connect_logs_and_traces.md (hunk 3 above)
Comment (lucaspimentel) [reply content/en/tracing/advanced/connect_logs_and_traces.md]: Thanks! 7a26d61db00436c90d86b2c0e12a2512171ab2b4

File: content/en/tracing/setup/dotnet.md (hunk 4 above)
Comment (lucaspimentel) [reply content/en/tracing/setup/dotnet.md]: Oops. 85a58fd59d34a22d28485de09d01d06b2dd4b383

**Comment by lucaspimentel** [comment]
@zacharycmontoya: can you provide text and sample core for the [Manual Trace ID injection](https://github.com/DataDog/documentation/pull/4844/files#diff-da66925c687a9d531438e62b6ee1ee9dR440) section?

File: content/en/tracing/advanced/connect_logs_and_traces.md [hunk 5]
```diff
@@ -119,6 +119,29 @@ This enables automatic trace ID injection for `winston`, `bunyan`, and `pino`.
 
 **Note**: Automatic injection only works for logs formatted as JSON.
 
+{{% /tab %}}
+{{% tab ".NET" %}}
+
+The .NET Tracer uses the [LibLog][2] library to automatically inject trace IDs into your application logs. It contains transparent built-in support for injecting into [NLog][3], [Log4Net][4], and [Serilog][5].
+
+Enable injection in the .NET Tracer’s [configuration][1] by setting `DD_LOGS_INJECTION=true` through environment variables or configuration files, or in code:
```
Comment (ruthnaebeck) [inline content/en/tracing/advanced/connect_logs_and_traces.md]:
```suggestion
Enable injection in the .NET Tracer’s [configuration][1] by setting `DD_LOGS_INJECTION=true` through environment variables or the configuration files. Additionally, injection can be enabled in the code:
```

File: content/en/tracing/setup/dotnet.md [hunk 6]
```diff
@@ -30,21 +30,22 @@ To begin tracing applications written in any language, first [install and config
 
 There are multiple ways to configure the .NET Tracer:
 
-- via code, using properties on `Tracer.Settings`
+- via code, using properties on the `TracerSettings` class
 - environment variables
 - `<appSettings>` section of the `app.config`/`web.config` file (.NET Framework only)
 
-The following table lists the supported configuration variables. Tracer settings can be accessed through properties on `Tracer.Settings`.
+The following table lists the supported configuration variables. The first column indicates the name used in environment variables or configuration files. The second column indicates the name of the propery on the `TracerSettings` class. You can access these properties in code through `Tracer.Instance.Settings`.
```
Comment (ruthnaebeck) [inline content/en/tracing/setup/dotnet.md]:
```suggestion
The following table lists the supported configuration variables. The first column indicates the name used in environment variables or configuration files. The second column indicates the name of the propery on the `TracerSettings` class. You can access these properties in the code through `Tracer.Instance.Settings`.
```

File: content/en/tracing/setup/dotnet.md [hunk 7]
```diff
@@ -286,21 +287,20 @@ ENV DD_INTEGRATIONS=/opt/datadog/integrations.json
 
 ### Configuration
 
-In addition to the settings listed in [Getting Started][10], the following tables list configuration variables specific to automatic instrumentation.
+In addition to the settings listed in [Getting Started][10], the following tables list configuration variables specific to automatic instrumentation. The first column indicates the name used in environment variables or configuration files. The second column indicates the name of the propery on the `TracerSettings` class. You can access these properties in code through `Tracer.Instance.Settings`.
```
Comment (ruthnaebeck) [inline content/en/tracing/setup/dotnet.md]:
```suggestion
In addition to the settings listed in [Getting Started][10], the following tables list configuration variables specific to automatic instrumentation. The first column indicates the name used in environment variables or configuration files. The second column indicates the name of the propery on the `TracerSettings` class. You can access these properties in the code through `Tracer.Instance.Settings`.
```

File: content/en/tracing/setup/dotnet.md [hunk 8]
```diff
@@ -286,21 +287,20 @@ ENV DD_INTEGRATIONS=/opt/datadog/integrations.json
...
-`DD_DISABLED_INTEGRATIONS`               | `DisabledIntegrationNames` | Sets a list of integrations to disable. All other integrations remain enabled. If not set, all integrations are enabled. Supports multiple values separated with semicolons. Valid values are the integration names listed in the [Integrations][3] section below. |
-`DD_TRACE_ANALYTICS_ENABLED`             | `AnalyticsEnabled`         | Shorthand that enables default Trace Search and Analytics settings for web framework integrations. Valid values are: `true` or `false` (default).                                                                                                                  |
+The following table lists integration-specific settings. The first column indicates the name used in environment variables or configuration files. The second column indicates the name of the propery on the `IntegrationSettings` class. You can access these properties in code through `Tracer.Instance.Settings.Integrations["<INTEGRATION>"]`. Integration names are listed in the [Integrations][3] section below.
```
Comment (ruthnaebeck) [inline content/en/tracing/setup/dotnet.md]:
```suggestion
The following table lists integration-specific settings. The first column indicates the name used in environment variables or configuration files. The second column indicates the name of the propery on the `IntegrationSettings` class. You can access these properties in the code through `Tracer.Instance.Settings.Integrations["<INTEGRATION>"]`. Integration names are listed in the [Integrations][3] section below.
```

File: content/en/tracing/advanced/connect_logs_and_traces.md [hunk 9]
```diff
@@ -119,6 +119,29 @@ This enables automatic trace ID injection for `winston`, `bunyan`, and `pino`.
 
 **Note**: Automatic injection only works for logs formatted as JSON.
 
+{{% /tab %}}
+{{% tab ".NET" %}}
+
+The .NET Tracer uses the [LibLog][2] library to automatically inject trace IDs into your application logs. It contains transparent built-in support for injecting into [NLog][3], [Log4Net][4], and [Serilog][5].
+
+Enable injection in the .NET Tracer’s [configuration][1] by setting `DD_LOGS_INJECTION=true` through environment variables or the configuration files. Additionally, injection can be enabled in the code:
```
Comment (andrewsouthard1) [inline content/en/tracing/advanced/connect_logs_and_traces.md]: Let's add a line break between these two thoughts and we're good to go.

Comment (lucaspimentel) [reply content/en/tracing/advanced/connect_logs_and_traces.md]: done!

**Pull request merged**